cluster/pid_cache.go

```diff
@@ -1,29 +1,116 @@
 package cluster
 
 import (
-	"sync"
+	"time"
 
 	"github.com/AsynkronIT/protoactor-go/actor"
+	"github.com/AsynkronIT/protoactor-go/remote"
+	"github.com/AsynkronIT/protoactor-go/router"
 )
 
-type pidCache struct {
-	lock  sync.RWMutex
-	Cache map[string]*actor.PID
+var (
+	pidCacheActorPid *actor.PID
+)
+
+func spawnPidCacheActor() {
+	props := router.NewConsistentHashPool(128).WithProducer(newPidCacheActor())
+	pidCacheActorPid, _ = actor.SpawnNamed(props, "PidCache")
+
+}
+func newPidCacheActor() actor.Producer {
+	return func() actor.Actor {
+		return &pidCachePartitionActor{}
+	}
+}
+
+type pidCachePartitionActor struct {
+	Cache        map[string]*actor.PID
+	ReverseCache map[string]string
 }
 
-func (c *pidCache) Get(key string) *actor.PID {
-	c.lock.RLock()
-	pid := c.Cache[key]
-	c.lock.RUnlock()
-	return pid
+type pidCacheResponse struct {
+	pid       *actor.PID
+	name      string
+	kind      string
+	respondTo *actor.PID
 }
 
-func (c *pidCache) Add(key string, pid *actor.PID) {
-	c.lock.Lock()
-	c.Cache[key] = pid
-	c.lock.Unlock()
+type pidCacheRequest struct {
```

Comment:
Similar to remote.ActorPidRequest, but tailored to work with the consistent hashrouter over the name field.